pkg/pidfile/pidfile.go

```diff
@@ -0,0 +1,50 @@
+package pidfile
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"strconv"
+)
+
+type PidFile struct {
+	path string
+}
+
+func checkPidFileAlreadyExists(path string) error {
+	if pidString, err := ioutil.ReadFile(path); err == nil {
+		if pid, err := strconv.Atoi(string(pidString)); err == nil {
+			if _, err := os.FindProcess(pid); err == nil {
```

Comment:
That's strange BTW. The create duplicate test passes on Mac OS, and it also
passes on Linux. It does however not with the `/proc` implementation
(`/proc` doesn't exist on Mac OS, apparently)...
On Feb 26, 2015 3:46 PM, "Rik Nijessen" <notifications@github.com> wrote:

> In pkg/pidfile/pidfile.go
> <https://github.com/docker/docker/pull/11018#discussion_r25431113>:
>
> > +import (
> > +	"fmt"
> > +	"io/ioutil"
> > +	"log"
> > +	"os"
> > +	"strconv"
> > +)
> > +
> > +type PidFile struct {
> > +	path string
> > +}
> > +
> > +func checkPidFileAlreadyExists(path string) error {
> > +	if pidString, err := ioutil.ReadFile(path); err == nil {
> > +		if pid, err := strconv.Atoi(string(pidString)); err == nil {
> > +			if _, err := os.FindProcess(pid); err == nil {
>
> It does not (it only works in Windows).
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/docker/docker/pull/11018/files#r25431113>.
>